src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php

```diff
@@ -24,9 +28,12 @@ class ControllerNameParser
 {
     protected $kernel;
 
-    public function __construct(KernelInterface $kernel)
+    private $layout;
+
+    public function __construct(KernelInterface $kernel, ControllerLayoutInterface $layout = null)
```

Comment:
As there is (currently) no configuration for this, how would someone change this to resolve it differently? Do you have examples of _why_ it should be solved differently?

Reply:
General idea is to reuse the class itself for other layouts. So we can use default generic layout for controllers and custom layouts with same logic for something else.

Am not sure that this class should live here, but not in http-kernel component (to be used standalone), but moving a class is a tricky thing, so I haven't suggest it here

Reply:
To be more concrete we have secondary routing for RPC controllers allowing to specify `a:b:c` notation which points to `RPC` namespace. Currently we have to copy-paste this class to override layout and keep the resolving process